zkvm/src/types.rs

```diff
@@ -329,6 +375,36 @@ impl Data {
             },
         }
     }
+
+    /// Encodes blinded Data values for txprogram bytecode.
+    pub fn encode(&self, program: &mut Vec<u8>) {
+        match self {
+            Data::Opaque(x) => {
+                program.append(&mut x.clone());
```

Comment:
We don't need to clone the opaque data vector here:
```suggestion
                program.extend_from_slice(&x);
```